[BUG] test_list_workloads Integration Test Points to Problematic OSB Workloads Commit

The `test_list_workloads` integration test attempts to load the available workloads from a specific commit hash: `000f2af53b9c34a2e83277e936dac113d497a80f`. The `challenges` directory had not been renamed to `test_procedures` [in that tree](https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/000f2af53b9c34a2e83277e936dac113d497a80f/so), causing the below error:

```
it/list_test.py::test_list_workloads[in-memory-it]
...

   ____                  _____                      __       ____                  __                         __
  / __ \____  ___  ____ / ___/___  ____ ___________/ /_     / __ )___  ____  _____/ /_  ____ ___  ____ ______/ /__
 / / / / __ \/ _ \/ __ \\__ \/ _ \/ __ `/ ___/ ___/ __ \   / __  / _ \/ __ \/ ___/ __ \/ __ `__ \/ __ `/ ___/ //_/
/ /_/ / /_/ /  __/ / / /__/ /  __/ /_/ / /  / /__/ / / /  / /_/ /  __/ / / / /__/ / / / / / / / / /_/ / /  / ,<
\____/ .___/\___/_/ /_/____/\___/\__,_/_/   \___/_/ /_/  /_____/\___/_/ /_/\___/_/ /_/_/ /_/ /_/\__,_/_/  /_/|_|
    /_/

[ERROR] Cannot list. Workload 'so' is invalid.

Error details: [] is too short
Instance: []
Path: deque(['test_procedures'])
Schema path: deque(['properties', 'test_procedures', 'minItems'])

Getting further help:
*********************
* Check the log files in /Users/engechas/.benchmark/logs for errors.
* Read the documentation at https://opensearch.org/docs.
* Ask a question on the forum at https://discuss.opendistrocommunity.dev/.
* Raise an issue at https://github.com/opensearch-project/OpenSearch-Benchmark/issues and include the log files in /Users/engechas/.benchmark/logs.

-------------------------------
[INFO] FAILURE (took 0 seconds)
-------------------------------
FAILED
```

--- a/it/list_test.py
+++ b/it/list_test.py
@@ -45,7 +45,7 @@ def test_list_opensearch_plugins(cfg):
 def test_list_workloads(cfg):
     assert it.osbenchmark(cfg, "list workloads") == 0
     assert it.osbenchmark(cfg, "list workloads --workload-repository=default "
-                           "--workload-revision=000f2af53b9c34a2e83277e936dac113d497a80f") == 0
+                           "--workload-revision=b28c9615d576bcc4beeb6725f4c95ed602b76638") == 0
 
 
 @it.benchmark_in_mem

